dev/com.ibm.ws.javaee.dd.common/src/com/ibm/ws/javaee/dd/common/ManagedExecutor.java

```diff
@@ -47,8 +47,25 @@ public interface ManagedExecutor extends JNDIEnvironmentRef, Describable {
      */
     boolean isSetMaxAsync();
 
+    /**
+     * @return &lt;virtual&gt; if specified
+     * @see #isSetVirutal
+     */
+    boolean isVirtual();
+
+    /**
+     * @return true if &lt;virtual&gt; is specified
+     * @see #isVirtual
+     */
+    boolean isSetVirtual();
+
+    /**
+     * @return &lt;qualifier&gt; elements as a read-only list
+     */
+    String[] getQualifier();
```

Comment:
The code would read better if we can make this and similar places plural:
`String[] getQualifiers();`
I don't know if that is allowed or if the names need to match exactly, but it looks like a similar pattern is used on the following lines using the plural form `getProperties` to obtain multiple `property` elements